### PR TITLE
Martin/ios 522 input height animation

### PR DESCRIPTION
**Pull request by MartinJibber**
https://user-images.githubusercontent.com/88675954/157577662-35e911b9-c1fc-48aa-b2c0-9bec808921fb.mp4

### CODE REVIEW

File: Jibber.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2112,8 +2114,9 @@
 				D6FE05212767B2BC00137EEA /* old_EmotionView.swift */,
 				D6BEDE4B2625D754009BA131 /* SwipeableInputAccessoryView.swift */,
 				8020DD92272C81FB00D5BA04 /* SwipeableInputAccessoryView.xib */,
-				80536B5D27A249D00094B81C /* SwipeInputPanGestureHandler.swift */,
+				8004E3D027D7C81600E9931F /* SwipeableInputAccessoryViewController.swift */,
```
Comment (MartinJibber) [inline Jibber.xcodeproj/project.pbxproj:2117]: The swipe input has been split into a view controller and a content view.

File: Shared/Managers/KeyboardManager.swift [hunk 2]
```diff
@@ -104,9 +104,12 @@ class KeyboardManager {
...
         var inputAccessoryHeight: CGFloat = 0
-        if let responder = UIResponder.firstResponder,
-           let inputAccessoryView = responder.inputAccessoryView {
-
+        guard let responder = UIResponder.firstResponder else {
+            return inputAccessoryHeight
+        }
+        if let inputAccessoryView = responder.inputAccessoryView {
+            inputAccessoryHeight = inputAccessoryView.height
+        } else if let inputAccessoryView = responder.inputAccessoryViewController?.view {
```
Comment (MartinJibber) [inline Shared/Managers/KeyboardManager.swift:112]: The keyboard manager wasn't set up to handle input view controllers. Fixed it.

File: iOS/Flows/Conversation/Conversation List/ConversationListViewController.swift [hunk 3]
```diff
@@ -36,20 +36,19 @@ class ConversationListViewController: ViewController, ConversationListCollection
...
-    lazy var messageInputAccessoryView: SwipeableInputAccessoryView = {
-        let inputView: SwipeableInputAccessoryView = SwipeableInputAccessoryView.fromNib()
-        inputView.delegate = self.swipeInputDelegate
-        inputView.textView.restorationIdentifier = "list"
-        return inputView
+    lazy var messageInputController: SwipeableInputAccessoryViewController = {
+        let inputController = SwipeableInputAccessoryViewController()
+        inputController.delegate = self.swipeInputDelegate
+        inputController.swipeInputView.textView.restorationIdentifier = "list"
+        return inputController
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Conversation List/ConversationListViewController.swift:43]: Pretty much the same as before except now we return a vc.

File: iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift [hunk 4]
```diff
@@ -10,48 +10,14 @@ import Foundation
 import Lottie
 import Combine
 
-protocol SwipeableInputAccessoryViewDelegate: AnyObject {
```
Comment (MartinJibber) [inline iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift:13]: I moved this delegate to the vc.

File: iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift [hunk 5]
```diff
@@ -64,86 +30,51 @@ class SwipeableInputAccessoryView: BaseView {
...
-    let deliveryTypeView = DeliveryTypeView()
-    let emotionView = old_EmotionView()
-
-    private var cancellables = Set<AnyCancellable>()
-
-    // MARK: - Layout/Animation Properties
+    // The input container and avatar height together determine the height of the whole view.
+    // When either of these two constraints are changed, the superview will resize to fix the new height.
+    @IBOutlet var inputContainerHeightConstraint: NSLayoutConstraint!
+    @IBOutlet var avatarHeightConstraint: NSLayoutConstraint!
```
Comment (MartinJibber) [inline iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift]: See comment.

File: iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift [hunk 6]
```diff
@@ -64,86 +30,51 @@ class SwipeableInputAccessoryView: BaseView {
...
-    let deliveryTypeView = DeliveryTypeView()
-    let emotionView = old_EmotionView()
-
-    private var cancellables = Set<AnyCancellable>()
-
-    // MARK: - Layout/Animation Properties
+    // The input container and avatar height together determine the height of the whole view.
+    // When either of these two constraints are changed, the superview will resize to fit the new height.
+    @IBOutlet var inputContainerHeightConstraint: NSLayoutConstraint!
+    @IBOutlet var avatarHeightConstraint: NSLayoutConstraint!
```
Comment (MartinJibber) [inline iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift:51]: See comment.

File: iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift [hunk 7]
```diff
@@ -64,86 +30,52 @@ class SwipeableInputAccessoryView: BaseView {
...
-        }
+        self.emotionView.configure(for: nil)
         
         self.inputTypeContainer.addSubview(self.deliveryTypeView)
         self.deliveryTypeView.alpha = 0
-        
-        self.deliveryTypeView.didSelectContext = { [unowned self] context in
-            AnalyticsManager.shared.trackEvent(type: .deliveryTypeSelected, properties: ["value": context.rawValue])
-            self.currentContext = context
-        }
```
Comment (MartinJibber) [inline iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift:137]: Moving input handling like this to the VC.

File: iOS/UI/Swipeable Input/SwipeableInputAccessoryViewController.swift [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+                                 withPreviewFrame frame: CGRect) -> Bool
+    /// The accessory view finished its swipe interaction.
+    func swipeableInputAccessory(_ controller: SwipeableInputAccessoryViewController,
+                                 didFinishSwipeSendingSendable didSend: Bool)
+
+    /// The avatar view in the accessory was tapped.
+    func swipeableInputAccessoryDidTapAvatar(_ controller: SwipeableInputAccessoryViewController)
+}
+
+class SwipeableInputAccessoryViewController: UIInputViewController {
```
Comment (MartinJibber) [inline iOS/UI/Swipeable Input/SwipeableInputAccessoryViewController.swift:33]: The VC handles input, while the view handles layout.

Most of this is just transplanted directly from the view.

**Review by Benjamindodgson — APPROVED** [review]

**Pull request merged**